BUILDING.md

```diff
@@ -1,11 +1,108 @@
+# Getting started / building from source
+
+1. clone the source code from the [J2V8 GitHub repository](https://github.com/eclipsesource/J2V8)
+2. run `j2v8-cli.cmd` (on Win32) or `source j2v8-cli.sh` on MacOS / Linux
+3. `nodejs git clone` to clone the Node.js/V8 source code
+4. `nodejs diff apply` to apply the required modifications to the Node.js source code
+5. start the desired J2V8 build either via `build -i` or `build ...args` (see below for details)
+
+# Build-System CLI
+
+## Interactive
+```shell
+build --i, --interactive
+# or
+python build.py --i, --interactive
+
+entering interactive mode...
+
+[0] Docker >> android-x86 >> NODE_ENABLED
+[1] Docker >> android-arm >> NODE_ENABLED
+[2] Docker >> alpine-linux-x64 >> NODE_ENABLED
+[3] Docker >> linux-x64 >> NODE_ENABLED
+[4] Docker >> linux-x86 >> NODE_ENABLED
+[5] Vagrant >> macosx-x64 >> NODE_ENABLED
+[6] Vagrant >> macosx-x86 >> NODE_ENABLED
+[7] Native >> windows-x64 >> NODE_ENABLED
+[8] Docker >> windows-x64 >> NODE_ENABLED
+[9] Vagrant >> windows-x64 >> NODE_ENABLED
+
+Select a predefined build-configuration to run: 2
+Building: Docker >> alpine-linux-x64 >> NODE_ENABLED
+
+Override build-steps ? (leave empty to run pre-configured steps): nodejs j2v8 test
+```
+
+## Non-interactive
+```shell
+build -h, --help
+# or
+python build.py -h, --help
+
+usage: build [-h] --target {android,linux,macos,win32} --arch {x86,x64,arm}
+             [--vendor VENDOR] [--keep-native-libs] [--node-enabled]
+             [--docker] [--vagrant] [--sys-image SYS_IMAGE] [--no-shutdown]
+             [--redirect-stdout] [--interactive]
+             [build-steps [build-steps ...]]
+```
+
+### Basic Examples
+
+Build for Alpine-Linux x64 using Docker and Node.js features included:<br/>
```

Comment:
Alpine is kinda exotic, I would put a documentation for a typical ubuntu user as first.